libraries/AP_HAL_ChibiOS/hwdef/common/stm32_util.c

```diff
@@ -229,29 +231,35 @@ void set_rtc_backup(uint8_t n, uint32_t v)
 #else
     PWR->CR1 |= PWR_CR1_DBP;
 #endif
-    ((__IO uint32_t *)&RTC->BKP0R)[n] = v;
+    while (n--) {
+        ((__IO uint32_t *)&RTC->BKP0R)[idx++] = *v++;
+    }
 }
 
 // see if RTC registers is setup for a fast reboot
 enum rtc_boot_magic check_fast_reboot(void)
 {
-    return (enum rtc_boot_magic)get_rtc_backup(0);
+    uint32_t v;
+    get_rtc_backup(0, &v, 1);
+    return (enum rtc_boot_magic)v;
```

Comment:
Shouldn't this be doing an `& RTC_BOOT_FAST`?  The values we poke into here are currently orthogonal, but if that changes these are going to cause issues.